src/errors/__tests__/network_error.test.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import NetworkError from "../network_error";
+import RequestError from "../request_error";
+
+describe("errors - NetworkError", () => {
+  it("should format an OtherError when called with minimal arguments", () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("foo", requestError);
+    expect(networkError).toBeInstanceOf(Error);
+    expect(networkError.name).toBe("NetworkError");
+    expect(networkError.type).toBe("NETWORK_ERROR");
+    expect(networkError.xhr).toBe(requestError.xhr);
+    expect(networkError.status).toBe(requestError.status);
+    expect(networkError.errorType).toBe(requestError.type);
+    expect(networkError.reason).toBe(requestError);
+    expect(networkError.code).toBe("");
+    expect(networkError.fatal).toBe(false);
+    expect(networkError.message).toBe("NetworkError () bar");
+  });
+
+  it("should be able to set it as fatal", () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "http://www.example.com");
+    const requestError = new RequestError(xhr, "foo", "bar");
+    const networkError = new NetworkError("foo", requestError, true);
+    expect(networkError).toBeInstanceOf(Error);
+    expect(networkError.name).toBe("NetworkError");
+    expect(networkError.type).toBe("NETWORK_ERROR");
+    expect(networkError.xhr).toBe(requestError.xhr);
+    expect(networkError.status).toBe(requestError.status);
+    expect(networkError.errorType).toBe(requestError.type);
+    expect(networkError.reason).toBe(requestError);
+    expect(networkError.code).toBe("");
+    expect(networkError.fatal).toBe(true);
+    expect(networkError.message).toBe("NetworkError () bar");
+  });
+
+  it("should filter in a valid error code", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 38. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=440&issues=AWgpL74vUMkE2J58feAY&open=AWgpL74vUMkE2J58feAY)

[ISSUE_KEY]: AWgpL74vUMkE2J58feAY